Build.cmake

```diff
@@ -101,35 +101,57 @@ ENDIF()
    ENDIF (MSVC OR MINGW)
 
    set(SRC_FILES ${SRC_FILES} ${SRC_PLATFORM_SPECIFIC})
-
-   # Create the g3log library
    INCLUDE_DIRECTORIES(${LOG_SRC})
-  SET(G3LOG_LIBRARY g3logger)
-
-   ADD_LIBRARY(${G3LOG_LIBRARY} SHARED ${SRC_FILES})
-   SET(${G3LOG_LIBRARY}_VERSION_STRING ${VERSION})
-   MESSAGE("Creating ${G3LOG_LIBRARY} VERSION: " ${VERSION})
-   SET_TARGET_PROPERTIES(g3logger PROPERTIES LINKER_LANGUAGE CXX SOVERSION ${VERSION})
-
-   set_target_properties(${G3LOG_LIBRARY} PROPERTIES
+   
+# Create the g3log static library
+   SET(G3LOG_STATIC_LIB g3logger_static)
+   ADD_LIBRARY(${G3LOG_STATIC_LIB} ${SRC_FILES})
+   SET_TARGET_PROPERTIES(${G3LOG_STATIC_LIB} PROPERTIES
       LINKER_LANGUAGE CXX
-      OUTPUT_NAME g3logger
+      OUTPUT_NAME g3logger_static
       CLEAN_DIRECT_OUTPUT 1)
+   TARGET_LINK_LIBRARIES(${G3LOG_STATIC_LIB} ${PLATFORM_LINK_LIBRIES})
+   TARGET_INCLUDE_DIRECTORIES(${G3LOG_STATIC_LIB} PUBLIC ${LOG_SRC})
+   
+   # Set current library for build examples and tests
+   # use default static, because the shared can not build below on this script
+   SET(G3LOG_LIBRARY ${G3LOG_STATIC_LIB})
+   
+# Create the g3log shared library
+IF( NOT(MSVC OR MINGW) OR 
+    (NOT ${ADD_BUILD_WIN_SHARED}.x STREQUAL ".x" AND MSVC AND NOT(CMAKE_VERSION LESS 3.4))
+  )
+   
+   SET(G3LOG_SHARED_LIB g3logger)
 
-   IF(APPLE)
-      set_target_properties(${G3LOG_LIBRARY} PROPERTIES MACOSX_RPATH TRUE)
-   ENDIF()
-
+   # set CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS before add_library
    IF(NOT ${ADD_BUILD_WIN_SHARED}.x STREQUAL ".x")
       IF(NOT(CMAKE_VERSION LESS 3.4) AND MSVC)
          set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
       ENDIF()
    ENDIF()
 
+   ADD_LIBRARY(${G3LOG_SHARED_LIB} SHARED ${SRC_FILES})
+   SET(${G3LOG_SHARED_LIB}_VERSION_STRING ${VERSION})
+   MESSAGE("Creating ${G3LOG_SHARED_LIB} VERSION: " ${VERSION})
+   SET_TARGET_PROPERTIES(g3logger PROPERTIES LINKER_LANGUAGE CXX SOVERSION ${VERSION})
 
-   TARGET_LINK_LIBRARIES(${G3LOG_LIBRARY} ${PLATFORM_LINK_LIBRIES})
+   SET_TARGET_PROPERTIES(${G3LOG_SHARED_LIB} PROPERTIES
+      LINKER_LANGUAGE CXX
+      OUTPUT_NAME g3logger
+      CLEAN_DIRECT_OUTPUT 1)
+
+   IF(APPLE)
+      SET_TARGET_PROPERTIES(${G3LOG_SHARED_LIB} PROPERTIES MACOSX_RPATH TRUE)
+   ENDIF()
+
+   TARGET_LINK_LIBRARIES(${G3LOG_SHARED_LIB} ${PLATFORM_LINK_LIBRIES})
 
    # Kjell: This is likely not necessary, except for Windows?
-   TARGET_INCLUDE_DIRECTORIES(${G3LOG_LIBRARY} PUBLIC ${LOG_SRC})
+   TARGET_INCLUDE_DIRECTORIES(${G3LOG_SHARED_LIB} PUBLIC ${LOG_SRC})
+   
+   # Set current library for build examples and tests
+   SET(G3LOG_LIBRARY ${G3LOG_SHARED_LIB})
```

Comment:
```${G3LOG_LIBRARY}``` should be either the static or the shared.  You can avoid building both by using the option approach.  That is a good way of validating the library since you can choose to build and test them with either one of the two types of library.